lib/ansible/module_utils/network/nxos/utils/telemetry/telemetry.py

```diff
@@ -0,0 +1,186 @@
+            module.fail_json(msg=msg)
+
+    if type == 'sensor_groups':
+        if not playvals.get('id'):
+            msg = "Invalid playbook value: {0}.".format(playvals)
+            msg = msg + " Parameter <id> under <sensor_groups> is required"
+            module.fail_json(msg=msg)
+        if playvals.get('path') and 'name' not in playvals['path'].keys():
+            msg = "Invalid playbook value: {0}.".format(playvals)
+            msg = msg + " Parameter <path> under <sensor_groups> requires <name> key"
+            module.fail_json(msg=msg)
+
+
+def get_instance_data(key, cr_key, cr, existing_key):
+    """
+    Helper method to get instance data used to populate list structure in config
+    fact dictionary
+    """
+    data = {}
+    if existing_key is None:
+        instance = None
+    else:
+        instance = cr._ref[cr_key]['existing'][existing_key]
+
+    if key == 'destination_groups':
+        m = re.search(r"destination-group (\d+)", cr._ref['_resource_key'])
+        instance_key = m.group(1)
+        data = {'id': instance_key, cr_key: instance}
+
+    if key == 'sensor_groups':
+        m = re.search(r"sensor-group (\d+)", cr._ref['_resource_key'])
+        instance_key = m.group(1)
+        data = {'id': instance_key, cr_key: instance}
+
+    if key == 'subscriptions':
+        m = re.search(r"subscription (\d+)", cr._ref['_resource_key'])
+        instance_key = m.group(1)
+        data = {'id': instance_key, cr_key: instance}
+
+    # Remove None values
+    data = dict((k, v) for k, v in data.items() if v is not None)
+    return data
+
+
+def cr_key_lookup(key, mo):
+    """
+    Helper method to get instance key value for mo
+    """
+    cr_keys = [key]
+    if key == 'destination_groups' and mo == 'TMS_DESTGROUP':
+        cr_keys = ['destination']
+    elif key == 'sensor_groups' and mo == 'TMS_SENSORGROUP':
+        cr_keys = ['data_source', 'path']
+    elif key == 'subscriptions' and mo == 'TMS_SUBSCRIPTION':
+        cr_keys = ['destination_group', 'sensor_group']
+
+    return cr_keys
+
+
+def normalize_data(cmd_ref):
+    ''' Normalize playbook values and get_exisiting data '''
+
+    playval = cmd_ref._ref.get('destination').get('playval')
+    existing = cmd_ref._ref.get('destination').get('existing')
+
+    dest_props = ['protocol', 'encoding']
+    if playval:
+        for prop in dest_props:
+            for key in playval.keys():
+                playval[key][prop] = playval[key][prop].lower()
+    if existing:
+        for key in existing.keys():
+            for prop in dest_props:
+                existing[key][prop] = existing[key][prop].lower()
+
+
+def remove_duplicate_context(cmds):
+    ''' Helper method to remove duplicate telemetry context commands '''
+    if not cmds:
+        return cmds
+    feature_indices = [i for i, x in enumerate(cmds) if x == "feature telemetry"]
+    telemetry_indeces = [i for i, x in enumerate(cmds) if x == "telemetry"]
```

Comment:
`indeces` is not a word 👎 

Ya got it right on the line above though. Interesting how you go back and forth with these. :-)